src/Bundle/Sender/Adapter/SymfonyMailerAdapter.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\MailerBundle\Sender\Adapter;
+
+use Sylius\Component\Mailer\Event\EmailSendEvent;
+use Sylius\Component\Mailer\Model\EmailInterface;
+use Sylius\Component\Mailer\Renderer\RenderedEmail;
+use Sylius\Component\Mailer\Sender\Adapter\AbstractAdapter;
+use Sylius\Component\Mailer\SyliusMailerEvents;
+use Symfony\Component\Mailer\MailerInterface;
+use Symfony\Component\Mime\Address;
+use Symfony\Component\Mime\Email;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+
+class SymfonyMailerAdapter extends AbstractAdapter
+{
+    /** @var MailerInterface */
+    protected $mailer;
+
+    /** @var EventDispatcherInterface|null */
+    protected $dispatcher;
+
+    public function __construct(MailerInterface $mailer, ?EventDispatcherInterface $dispatcher = null)
```

Comment:
As this is a new class, we can force its dependency. 
```suggestion
    public function __construct(MailerInterface $mailer, EventDispatcherInterface $dispatcher)
```

Reply:
As you want but EventDispatcherInterface is also nullable on DefaultAdapter.

Reply:
That's a good point, and should be fixed there as well. But that could be a topic for another PR.